SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -1295,7 +1307,58 @@ void USpatialActorChannel::ResetShadowData(FRepLayout& RepLayout, FRepStateStati
 	}
 }
 
-bool USpatialActorChannel::SatisfiesSpatialPositionUpdateRequirements(FVector& OutNewSpatialPosition)
+void USpatialActorChannel::CheckForClientEntityInterestUpdate()
+{
+	// Only valid if rep graph enabled, client entity interest is enabled, and we're processing a player controller
+	const double CurrentTime = NetDriver->GetElapsedTime();
+	const USpatialGDKSettings* Settings = GetDefault<USpatialGDKSettings>();
+	const UReplicationGraph* RepGraph = Cast<USpatialReplicationGraph>(NetDriver->GetReplicationDriver());
+	USpatialNetConnection* NetConnection = Cast<USpatialNetConnection>(Actor->GetNetConnection());
+	UNetReplicationGraphConnection* RepGraphConnection =
+		Cast<UNetReplicationGraphConnection>(NetConnection->GetReplicationConnectionDriver());
+
+	if (RepGraph == nullptr || RepGraphConnection == nullptr || NetConnection == nullptr)
+	{
+		UE_LOG(LogSpatialActorChannel, Error,
+			   TEXT("Failed to handle client entity interest, some connection or setting was misconfigured"));
+		return;
+	}
+
+	// If interest is already marked dirty (e.g. because we just gained authority over the PC and want to flush immediate),
+	// then don't bother doing any checks.
+	if (NetDriver->ActorSystem->IsClientInterestDirty(EntityId))
+	{
+		NetConnection->TimeWhenClientInterestLastUpdated = CurrentTime;
+		return;
+	}
+
+	bool bShouldMarkInterestDirty = false;
+
+	// Round robin updating client interest
+	if (RepGraph->GetReplicationGraphFrame() % Settings->ClientEntityIdInterestUpdateFrameFrequency
+		== RepGraphConnection->ConnectionOrderNum % Settings->ClientEntityIdInterestUpdateFrameFrequency)
+	{
+		bShouldMarkInterestDirty = true;
+	}
+
+	if (!bShouldMarkInterestDirty)
+	{
+		return;
+	}
+
+	if (UMetricsExport* MetricsExport = Actor->GetWorld()->GetGameInstance()->GetSubsystem<UMetricsExport>())
+	{
+		const FString ClientIdentifier = FString::Printf(TEXT("PC-%lld"), NetConnection->GetPlayerControllerEntityId());
+		const float TimeSinceLastClientInterestUpdate = CurrentTime - NetConnection->TimeWhenClientInterestLastUpdated;
```

Comment:
```suggestion
		const double TimeSinceLastClientInterestUpdate = CurrentTime - NetConnection->TimeWhenClientInterestLastUpdated;
```

Reply:
It ends up in here `TMap<FString, TMap<FString, float>> MetricsMap;`, do you want this take contain a double as well perhaps? 
Or if this downcast is intentional would prefer a `static_cast`.

Reply:
float is plenty of precision, will add a cast